src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringConcat.cs

```diff
@@ -195,7 +197,7 @@ private bool TryExtractStringConcatArgs(BoundExpression lowered, out ImmutableAr
                         // attempting to access its value as a string.
 
                         var rightConstant = boundCoalesce.RightOperand.ConstantValue;
-                        if (rightConstant != null && rightConstant.IsString && rightConstant.StringValue.Length == 0)
+                        if (rightConstant != null && rightConstant.IsString && rightConstant.StringValue!.Length == 0)
```

Comment:
how do we know?

Reply:
Because we just checked that the rightConstant is a string.

Reply:
Re-activating. What was the resolution here?

---
In reply to: [387364854](https://github.com/dotnet/roslyn/pull/42068#discussion_r387364854) [](ancestors = 387364854)

Reply:
In the expression `rightConstant.IsString && rightConstant.StringValue!.Length == 0` we already tested `IsString`, which ensures `StringValue` is not null.  There is no way to express that in the language yet, but the implication is completely local.

Reply:
Ah, yes!
Could leave a comment linking to https://github.com/dotnet/roslyn/issues/41964 (use MemberNotNull when available).